### DISCUSSION TITLE
Getting host in getServerSideProps using req.headers.host

### DISCUSSION THREAD

**Discussion by mmalomo** [vercel/next.js#58503] [Help]
### Summary

I have a production application that can be accessed from different domains. Depending on the domain from which the application is accessed, it loads different environment variables.

For example, on the login page (using the Firebase token as an example, but there are various environment variables related to domains as they belong to different clients):

```
function getServerSideProps({ req, res, locale }: GetServerSidePropsContext) {

  let tokenFromEnvVariable;

  if (req.headers.host == 'domain-one.com') {
    // Read Firebase token from environment variable
    tokenFromEnvVariable = process.env.TOKEN_1;
  } else if (req.headers.host == 'domain-two.com') {
    // Read Firebase token from environment variable
    tokenFromEnvVariable = process.env.TOKEN_2;
  }

  return {
    props: {
      firebaseToken: tokenFromEnvVariable
    }
  }

}
```

So far, everything works fine. Depending the domain used to access the app, the page receives the corresponding token on the client side. This is running the app on Node.js 16.x.

When I updated the runtime to Node.js `18.18.2`, `req.headers.host` no longer contains the host, and instead, it contains the IPv6 address. This is due to a change made in Node.js 17 where IPv6 is given preference.

In my case, it is crucial to know the domain from which the request is being made. Can you suggest a solution that does not involve sticking to Node.js 16.x forever?

### Additional information

I've tried `cross-env NODE_OPTIONS=--dns-result-order=ipv4first next dev` but with this `req.headers.host` is `127.0.0.1:XXXX` and what I'm looking for is `localhost` or `some-domain.com`. Just like in node 16

Same issue but with no answer in SO
https://stackoverflow.com/questions/77399249/next-js-13-next-headers-returning-strange-value-for-host

**Comment by PabloMarchesini** [comment]
Same issue here with the host header, couldn't find a solution yet!

**Comment by theCompanyDream** [comment]
I've been having the same issue seems like I can't dynamically generate the domain name for a single website. Did any of you eventually find an answer to this?
